src/drive.py

```diff
@@ -1,10 +1,79 @@
+# import db
+from Google import connect_to_drive
+import pandas as pd
 import db
 
+
+def get_folder_contents(service, id):
+
+    query = f"parents = '{id}'"
+    if not service:
+        print("Connection failed")
+        return []
+
+    response = service.files().list(q=query).execute()
+    files = response.get('files', [])
+    nextPageToken = response.get('nextPageToken')
+
+    while nextPageToken:
+        response = service.files().list(q=query, pageToken=nextPageToken).execute()
+        files.extend(response.get('files'))
+        nextPageToken = response.get('nextPageToken')
+
+    return files
+
+
+def list_drive(service, folder_id, result, parentName='', grandParentName=''):
+    files = get_folder_contents(service, folder_id)
+
+    for item in files:
+        if item['mimeType'] != 'application/vnd.google-apps.folder':
+            try:
+                id = item['id']
+                link = (service.files().get(fileId=id, fields='webViewLink').execute())[
+                    'webViewLink']
+                item['tags'] = [parentName, grandParentName]
```

Comment:
we need more than 2 ancestors in many cases

Reply:
did nothing